docs/reference/ilm/ilm-with-existing-indices.asciidoc

```diff
@@ -0,0 +1,409 @@
+start to finish, it may be useful to use {ilm-init} with existing indices,
+particularly when transitioning from an alternative method of managing the index
+lifecycle such as Curator, or when migrating from daily indices to
+rollover-based indices. Such use cases are fully supported, but there are some
+configuration differences from when {ilm-init} can manage the complete index
+lifecycle.
+
+This section describes strategies to leverage {ilm-init} for existing periodic
+indices when migrating to fully {ilm-init}-manged indices, which can be done in
+a few different ways, each providing different tradeoffs. As an example, we'll
+walk through a use case of a very simple logging index with just a field for the
+log message and a timestamp.
+
+First, we need to create a template for these indices:
+
+[source,js]
+-----------------------
+PUT _template/mylogs_template
+{
+  "index_patterns": [
+    "mylogs-*"
+  ],
+  "settings": {
+    "number_of_shards": 1,
+    "number_of_replicas": 1
+  },
+  "mappings": {
+    "properties": {
+      "message": {
+        "type": "text"
+      },
+      "@timestamp": {
+        "type": "date"
+      }
+    }
+  }
+}
+-----------------------
+// CONSOLE
+// TEST
+
+And we'll ingest a few documents to create a few daily indices:
+
+[source,js]
+-----------------------
+POST mylogs-pre-ilm-2019.06.24/_doc
+{
+  "@timestamp": "2019-06-24T10:34:00",
+  "message": "this is one log message"
+}
+-----------------------
+// CONSOLE
+// TEST[continued]
+
+[source,js]
+-----------------------
+POST mylogs-pre-ilm-2019.06.25/_doc
+{
+  "@timestamp": "2019-06-25T17:42:00",
+  "message": "this is another log message"
+}
+-----------------------
+// CONSOLE
+// TEST[continued]
+
+Now that we have these indices, we'll look at a few different ways of migrating
+these indices to ILM.
+
+[[ilm-with-existing-periodic-indices]]
+=== Managing existing periodic indices with {ilm-init}
+
+NOTE: The examples in this section assume daily indices as set up in
+<<ilm-with-existing-indices,the previous section>>.
+
+The simplest way to manage existing indices while transitioning to fully
+{ilm-init}-managed indices is to allow all new indices to be fully managed by
+{ilm-init} before attaching {ilm-init} policies to existing indices. To do this,
+all new documents should be directed to {ilm-init}-managed indices - if you are
+using Beats or Logstash data shippers, upgrading all of those shippers to
+version 7.0.0 or higher will take care of that part for you. If you are not
+using Beats or Logstash, you may need to set up ILM for new indices yourself as
+demonstrated in the <<getting-started-index-lifecycle-management,getting started
+guide>>.
+
+NOTE: If you are using Beats through Logstash, you may need to change your
+Logstash output configuration and invoke the Beats setup to use ILM for new
+data.
+// TODO: As soon as there's docs for Beats+LS to link to here, link to them.
```

Comment:
I'm not sure if we should merge this with a TODO in it?

Reply:
Thanks, I originally put that in there hoping that page would exist and kinda forgot about it. That page does not seem to exist yet, so I'll just remove this.